nexusutils/dataconverter/readers/xps/reader.py

```diff
@@ -0,0 +1,405 @@
+
+    for entry, xr_data in entries_values.items():
+        entry_set.add(entry)
+
+        chan_count = "_chan"
+
+        # Iteration over scan
+        for data_var in xr_data.data_vars:
+
+            if chan_count in data_var:
+                detector_num = data_var.split("_chan")[-1]
+                detector_nm = f"detector{detector_num}"
+                DETECTOR_SET.add(detector_nm)
+                scan_num = data_var.split("_scan")[-1].split("_chan")[0]
+                scan_nm = f"scan_{scan_num}"
+                modified_key = key.replace("entry", entry)
+                modified_key = modified_key.replace("[detector]", f"[{detector_nm}]")
+                modified_key = modified_key.replace("[data]", f"[{scan_nm}]")
+                modified_key_unit = modified_key + "/@units"
+
+                template[modified_key] = xr_data[data_var].data
+                key_indv_chan_sginal = modified_key.replace("/raw", "/@signal")
+                template[key_indv_chan_sginal] = "raw"
+                template[modified_key_unit] = config_dict[f"{key}/@units"]
+
+
+def fill_template_with_xps_data(config_dict,
+                                xps_data_dict,
+                                template,
+                                entry_set):
+    """Collect the xps data from xps_data_dict
+        and store them into template. We use searching_keys
+        for separating the data from xps_data_dict.
+    """
+    for key, value in config_dict.items():
+        if XPS_DATA_TOCKEN in value:
+            key_part = value.split(XPS_DATA_TOCKEN)[-1]
+            entries_values = find_entry_and_value(xps_data_dict,
+                                                  key_part,
+                                                  dt_typ=XPS_DATA_TOCKEN)
+
+            fill_data_group(key, entries_values, config_dict, template, entry_set)
+
+        if XPS_DETECTOR_TOCKEN in value:
+            key_part = value.split(XPS_DATA_TOCKEN)[-1]
+            entries_values = find_entry_and_value(xps_data_dict,
+                                                  key_part,
+                                                  dt_typ=XPS_DETECTOR_TOCKEN)
+
+            fill_detector_group(key, entries_values, config_dict, template, entry_set)
+
+        if XPS_TOCKEN in value:
+            tocken = value.split(XPS_TOCKEN)[-1]
+            entries_values = find_entry_and_value(xps_data_dict,
+                                                  tocken,
+                                                  dt_typ=XPS_TOCKEN)
+            for entry, ent_value in entries_values.items():
+                entry_set.add(entry)
+                modified_key = key.replace("[entry]", f"[{entry}]")
+                template[modified_key] = ent_value
+                try:
+                    template[f"{modified_key}/@units"] = config_dict[f"{key}/@units"]
+                except KeyError:
+                    pass
+
+
+def extract_atom_types(formula: str) -> Set:
```

Comment:
I think you should also check against valid element names, as @mkuebach does in the apm reader.